### ISSUE TITLE
ResourceIds null

### ISSUE THREAD

**Issue by andrecmilano** [fullcalendar/fullcalendar-scheduler#394]
Hello,

I recently started to use fullCalendar for a project, and so far I am really happy with it.

I integrated the scheduler view in my project. Based on what I understood from the documentation, the event and the resources are linked through the resourceId field in the object event. 
I do not have the resourceId value saved in the data_base, indeed I want to compute it at runtime. 

when I load the events I do:
   
```
eventSources: [
        {
            url: ENV.queryEventsUrl,

            type: 'POST',       

            success: function(events)
            {

                $.each(events, function(i, event)
                {

                    event.resourceIds = [];
                    for (o in event.occupations)
                    {

                      event.resourceIds.push("rooms"+event.occupations[o].room_id);

                      for (u in event.occupations[o].utilizations) {
                                event.resourceIds.push("resources"+event.occupations[o].utilizations[u].resource_id);}

                      for (p in event.occupations[o].partecipations) {
                                 event.resourceIds.push("users"+event.occupations[o].partecipations[p].user_id);}
                    }
                });
            }
        }
        ],

````

I manually set all the resourceIds in the event objects and also I assign the id to the resources manually:

```
        currentResources = ENV[resourceType]
            .filter(function(r) {
              return r.name.toLowerCase().includes(resourceFilter);
            })
            .map(function(r) {
              return { id: resourceType+r.id, title: r.name };
            });
```

The resources are correctly loaded:
![image](https://user-images.githubusercontent.com/17096956/34116527-2ea0dde4-e419-11e7-9031-57778b4fcd31.png)

However when I call: $('#calendar').fullCalendar( 'clientEvents') 
The result is:
![image](https://user-images.githubusercontent.com/17096956/34116576-552940f0-e419-11e7-8cd6-57a47b9147e2.png)

I tried many possible solutions to change the field in the event but I didn't succeed.

I think I am missing some simple setting variable that would allow me to modify the event and insert the correct resourceId. I checked online, in the documentation and in other issues on github but I was not able to find the same problem.

I hope the issue is clear enough. Please let me know if I need to provide additional info to clarify it.

Thank you for your help in advance.

**Comment by acerix** [comment]
Would you be able to post a [runnable, stripped-down demonstration of the bug](http://fullcalendar.io/wiki/Reduced-Test-Cases/)? Would really appreciate it because the time saved reproducing will be time spent fixing.

I tried to quickly replicate the issue, but `resourceIds` keeps the same values that are set in `eventSources` in my test:

https://codepen.io/anon/pen/VyeoNZ?editors=0010

**Comment by andrecmilano** [comment]
Hello,

Thank you for your answer is really precious.
I wanted to post a  runnable, stripped-down demonstration of the bug, however, it is complicated to replicate due to the architecture of the software which involves multiple methods defined in different files.


               
       var c = $e.calendar.fullCalendar({
       
        schedulerLicenseKey: 'CC-Attribution-NonCommercial-NoDerivatives',

        eventDrop: handleDragOrResize,
        eventResize: handleDragOrResize,
        eventClick: this.showEventPopup,
        // eventMouseover: function(event) { },
        // eventMouseout:  function(event) { },
        selectAllow: function()
        {
            var view = c.fullCalendar('getView').name;
            return view == 'agendaWeek' || view == 'timelineDay';
        },
        select:  function(start, end, jsEvent, view, resource)
        {
            if (resource) console.log("Popup with resource:", resourceType, '#',resource.id, resource);

            self.showEventPopup({
                start: start,
                end: end
            });
        },
        header: {
            left: 'prev,next,today',
            center: 'title',
            right: 'listMonth, month, agendaWeek, timelineDay',
        },

        resources: function(callback) {
          callback(currentResources);
        },
        viewRender: updateFilters,
        eventRender: function ( event, element, view ) {
            return !eventFilter || (event.title || "").toLowerCase().includes(eventFilter);
        },
        defaultView: 'agendaWeek',
        navLinks: true,
        editable: true,
        selectable: true,
        selectHelper: true,
        allDaySlot: false,
        nowIndicator: true,
        slotDuration: '0:15:00',
        slotLabelInterval: '1:00:00',
        scrollTime: '08:00',
        businessHours: {
            start: '08:00',
            end: '18:00'
        },

        eventSources: [
        {
            url: ENV.queryEventsUrl,
            type: 'POST',
            success: function(events)
            {
                $.each(events, function(i, event)
                {
                    event.resourceIds = [];
                    for (o in event.occupations) {
                          var resource_id = "Room"+event.occupations[o].room_id;
                          $e.calendar.fullCalendar('addResource', {
                            id: resource_id,
                            title: "Room"+event.occupations[o].room_id
                          });
                          event.resourceIds.push(resource_id);
                          for (u in event.occupations[o].utilizations)
                          {
                            var resource_id = "Resource"+event.occupations[o].utilizations[u].resource_id;
                            $e.calendar.fullCalendar('addResource', {
                              id: resource_id,
                              title: "Resource"+event.occupations[o].utilizations[u].resource_id
                            });
                            event.resourceIds.push(resource_id);
                          }
                          for (p in event.occupations[o].participations)
                          {
                            var resource_id = "User"+event.occupations[o].participations[p].user_id;
                            $e.calendar.fullCalendar('addResource', {
                              id: resource_id,
                              title: "User"+event.occupations[o].participations[p].user_id
                            });
                            event.resourceIds.push(resource_id);
                          }
                    }
                    if (eventColorization == 'status')
                    {
                        event.color = '#bbb';
                    }
                    else if (eventColorization == 'duration')
                    {
                        event.color = '#bbb';
                    }
                    else
                    {
                      if (!event.color) event.color = '#444';
                    }
                    event.textColor = $.Color(event.color).isDark() ? 'white' : 'black';
                });
            }
        }
        ],
    });



this is my fullCalendar function. When I tried to import in your runnable the different settings keeping the same eventResource function, your runnable was still working. 

I noticed something really weird during the runtime in my events list. During the eventResources function, it is assigned correctly a ResourceIds vector containing certain values. 
Here is a screenshot:

![image](https://user-images.githubusercontent.com/17096956/34253171-c79746f8-e646-11e7-941f-62c36f9d2325.png)

however, immediately after during the EventRender function, the resourceId and resourceIds are set to null. For example the event.textCOlor set at the end of EventSource is correctly set even in the next step and its value never change.
I do not understand why ResourceIds is overwritten somewhere. 
Here is the screenshot of the same event when EventRender is triggered:

![image](https://user-images.githubusercontent.com/17096956/34253343-67e28d84-e647-11e7-8f0f-446d80cee047.png)


I am sorry to not be able to write a runnable that represent the same issue. As I told you if I write the same configuration I have, but simplified (semplified = removed all the parameters related with functions that I have in my local js and that do not modify the events object), with codepen I get this:
https://codepen.io/andrecmilano/pen/eyzaVj

which actually works. But locally still the problem I described above is present.
I am really confused.

Thank you for your help.
I hoped this error occured already to other users. 
Maybe I still miss something in the configuration of FullCalendar. Not sure.

**Comment by andrecmilano** [comment]
Hello,

I found the error. 
It is in the function update filter and in the callback of resources.
I will look into them carefully and as soon as I solve them or I need more help I will post again on this issue.

Thank you again for your help.
:)

**Comment by acerix** [comment]
Thanks for the update, closing for now.

**Comment by andrecmilano** [comment]
Hello,

The error I reported above did not solve the problem. Or better the event keep the resourceIds value at null. I noticed that you released a new version of scheduler 1.9.1. I added it to my project, maybe this is the reason why I got the error before. However, when I use the new scheduler.js file I get this error:

![image](https://user-images.githubusercontent.com/17096956/34384564-337ef7be-eb1d-11e7-8f57-1462c9eda038.png)

Did it happen before to someone else? 
Do you know the cause of it?

It is pretty weird because I also tried to use the remote version of the js file (https://fullcalendar.io/js/fullcalendar-scheduler-1.9.1/scheduler.js) and I got the same error, indeed maybe I am missing something in my local configuration. May you help with this error?

Thank you

**Comment by acerix** [comment]
I can't tell what the issue is from the screenshot, you need to isolate the problem on your side.

https://fullcalendar.io/wiki/Reduced-Test-Cases/

It might help to paste your code in the CodePen template to see if you can replicate the issue there.

PS. Linking directly to the libraries on fullcalendar.io might cause an error, check the download page for CDN links.
